pandas/tests/test_window.py

```diff
@@ -667,6 +667,21 @@ def test_rolling_axis(self, axis_frame):
         result = df.rolling(3, axis=axis_frame).sum()
         tm.assert_frame_equal(result, expected)
 
+    def test_rolling_axis_count(self, axis_frame):
+        # see gh-26055
+        df = DataFrame({'x': range(3), 'y': range(3)})
+
+        axis = df._get_axis_number(axis_frame)
+
+        if axis == 0:
```

Comment:
this is not correct (should your test have broken?)

``if axis in [0, 'index']`` is what you need